baseframe/forms/auto.py

```diff
@@ -17,25 +18,25 @@ class ConfirmDeleteForm(Form):
     cancel = SubmitField(__(u"Cancel"))
 
 
-def render_form(form, title, message='', formid='form', submit=__(u"Submit"), cancel_url=None, ajax=False):
+def render_form(form, title, message='', formid='', submit=__(u"Submit"), cancel_url=None, ajax=False, with_chrome=True):
     multipart = False
+    if not formid:
+        formid = 'form-' + buid()
     for field in form:
         if isinstance(field.widget, wtforms.widgets.FileInput):
             multipart = True
-    if form.errors:
-        code = 200  # 400
-    else:
-        code = 200
-    if request.is_xhr and ajax:
+    if not with_chrome:
         template = THEME_FILES[current_app.config['theme']]['ajaxform.html.jinja2']
-        return make_response(render_template(template, form=form, title=title,
+        return render_template(template, form=form, title=title,
             message=message, formid=formid, submit=submit,
-            cancel_url=cancel_url, multipart=multipart), code)
+            cancel_url=cancel_url, ajax=ajax, multipart=multipart)
+    if request.is_xhr and ajax:
```

Comment:
Why do we need to check for both `request.is_xhr` and `ajax`? `request.is_xhr` should suffice?

Reply:
This was check that was previously there.
@jace will just `request.is_xhr` suffice?

Reply:
There are two considerations here:

1. This is an Ajax request (`request.is_xhr` is True)
2. The developer wanted this to be an Ajax form (`ajax` is True)

The developer may not want an Ajax form for reasons including (a) the page being simple enough to afford a full reload, and (b) testing the business logic with the interactive debugger.